STCommon/src/commonMain/kotlin/com/infomaniak/multiplatform_swisstransfer/common/interfaces/ui/FileUi.kt

```diff
@@ -24,19 +24,42 @@ import com.infomaniak.multiplatform_swisstransfer.common.interfaces.upload.Uploa
 data class FileUi(
     val uid: String,
     val fileName: String,
+    val isFolder: Boolean? = false,
     val fileSize: Long,
     val mimeType: String?,
     val localPath: String?,
+    var parent: File? = null,
+    var children: MutableList<File> = mutableListOf(),
 ) {
 
     constructor(file: File) : this(
         uid = file.uuid,
         fileName = file.fileName,
+        isFolder = file.isFolder,
         fileSize = file.receivedSizeInBytes,
         mimeType = file.mimeType,
         localPath = null,
     )
 
+    // Init for folder
+    constructor(folderName: String) : this(
+        uid = folderName,
```

Comment:
By doing this, `uid` won't be Unique IDs anymore.

Is it ok ?

Reply:
It should be ok but you're right. Just in case, I'll use a random UUID.